modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/poll/PollingSourceWrapper.java

```diff
@@ -204,6 +217,25 @@ private int compareWatermarks(Serializable w1, Serializable w2, Comparator compa
     return comparator.compare(w1, w2);
   }
 
+  @Override
+  public Map<String, Object> beginRestart() {
+    Map<String, Object> restartingContext = new HashMap<>();
+
+    restarting.set(true);
+
+    restartingContext.put(POLLING_SOURCE_EXECUTOR_KEY, executor);
+    restartingContext.put(RUNNABLE_KEY, delegateRunnable);
+    return restartingContext;
+  }
+
+  @Override
+  public void finishRestart(Map<String, Object> restartingContext) {
+    restarting.set(true);
+
+    executor = (org.mule.runtime.api.scheduler.Scheduler) restartingContext.get(POLLING_SOURCE_EXECUTOR_KEY);
+    delegateRunnable = (DelegateRunnable) restartingContext.get(RUNNABLE_KEY);
```

Comment:
although I insist this is not necesary, this isn't a bad idea as the delegate you're keeping from the prior instance is quite likely to have references to the old instance...potentially producing leaks